src/aap_eda/api/views/user.py

```diff
@@ -236,3 +240,12 @@ def get_serializer_class(self):
 
     def get_response_serializer_class(self):
         return serializers.UserDetailSerializer
+
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+
+        if instance == request.user:
```

Comment:
@msmagnanijr Should we check if the user is admin or not? Only admin has the right to delete, right?

Reply:
@hsong-rh From what I understand, not even the admin can delete itself. What do you think? @Alex-Izquierdo

Reply:
The title **Admin users can delete themselves in EDA** tells me admin can do it 😄

Reply:
@hsong-rh you have a point :sweat_smile:

Reply:
@msmagnanijr @hsong-rh the original bug was raised concerning an admin user, but in later discussions we decided to prevent it for any user. The UI will apply a similar restriction, disabling the button.